docs/tutorial/quickstart.md

```diff
@@ -0,0 +1,355 @@
+go: added golang.org/x/sync v0.6.0
+go: added oras.land/oras-go/v2 v2.5.0
+```
+
+Run the code.
+```
+go run .
+```
+
+You should see the tags in the repository.
+
+## Push a layer to the repository
+
+All referenced layers must exist in the repository before a manifest can be pushed. The following code snippet demonstrates how to push a manifest layer using the [(*Repository) Push](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.Push) method. Paste the code into `main.go` after the last section.
+
+```
+// 3. push a layer to the repository
+layer := []byte("example manifest layer")
+layerDescriptor := content.NewDescriptorFromBytes(v1.MediaTypeImageLayer, layer)
+err = repo.Push(ctx, layerDescriptor, bytes.NewReader(layer))
+if err != nil {
+	panic(err)
+}
+fmt.Println("Pushed manifest layer:", layerDescriptor.Digest)
+```
+
+## Push a manifest to the repository with the tag "quickstart"
+
+The following code snippet demonstrates how to pack a manifest and push it to the repository with the tag "quickstart" using the [PackManifest](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0#PackManifest) and the [(*Repository) Tag](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.Tag) methods. Paste the code into `main.go` after the last section.
+
+```
+// 4. Push a manifest to the repository with the tag "quickstart"
+packOpts := oras.PackManifestOptions{
+	Layers: []v1.Descriptor{layerDescriptor},
+}
+artifactType := "application/vnd.example+type"
+desc, err := oras.PackManifest(ctx, repo, oras.PackManifestVersion1_1, artifactType, packOpts)
+if err != nil {
+	panic(err)
+}
+tag := "quickstart"
+err = repo.Tag(ctx, desc, tag)
+if err != nil {
+	panic(err)
+}
+fmt.Println("Pushed and tagged manifest")
+```
+
+## Fetch the manifest from the repository by tag
+
+The following code snippet demonstrates how to fetch a manifest from the repository by its tag. First, the descriptor associated with the tag is resolved using [(*Repository) Resolve](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.Resolve). Then, the content is fetched and read using [FetchAll](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/content#FetchAll) from the [content](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/content) package. Paste the code into `main.go` after the last section.
+
+```
+// 5. Fetch the manifest from the repository by tag
+desc, err = repo.Resolve(ctx, tag)
+if err != nil {
+	panic(err)
+}
+fetchedManifestContent, err := content.FetchAll(ctx, repo, desc)
+if err != nil {
+	panic(err)
+}
+fmt.Println(string(fetchedManifestContent))
+```
+
+## Parse the fetched manifest content and get the layers
+The following code snippet demonstrates how to parse the fetched manifest content and get the layers. Paste the code into `main.go` after the last section.
+
+```
+// 6. Parse the fetched manifest content and get the layers
+var manifest v1.Manifest
```

Comment:
What's `v1`? It is new in this doc. You need to mention that we need a new import entry.

Reply:
Also, it is better to name it `ocispec` instead of `v1`.

Reply:
After adding the code, you should run `go mod tidy` again. It will mark the direct and indirect dependencies properly.